staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -615,8 +618,16 @@ func (c completedConfig) New(name string, delegationTarget DelegationTarget) (*G
 	const priorityAndFairnessConfigConsumerHookName = "priority-and-fairness-config-consumer"
 	if s.isPostStartHookRegistered(priorityAndFairnessConfigConsumerHookName) {
 	} else if c.FlowControl != nil {
-		err := s.AddPostStartHook(priorityAndFairnessConfigConsumerHookName, func(context PostStartHookContext) error {
-			go c.FlowControl.Run(context.StopCh)
+		err := s.AddPostStartHook(priorityAndFairnessConfigConsumerHookName, func(ctx PostStartHookContext) error {
+			loopBackClient, err := kubernetes.NewForConfig(ctx.LoopbackClientConfig)
+			if err != nil {
+				return fmt.Errorf("failed constructing pre-flight-check client for FlowControl config controller: %v", err)
+			}
+			_, err = loopBackClient.FlowcontrolV1alpha1().PriorityLevelConfigurations().List(context.Background(), metav1.ListOptions{})
```

Comment:
a) Put these lines in a named function ("ensureAPIRegistered") so people understand the purpose
b) Is it possible for this to be flaky? Can this get called before the api is being served, for example? I'm a little nervous about this.
c) Wouldn't it be simpler to check discovery rather than try to list? Also, less chance of a permissions problem?